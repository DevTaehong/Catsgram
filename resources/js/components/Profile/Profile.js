import React, {useState} from 'react';
import { createRoot } from 'react-dom/client';


function Profile({ userName, userEmail, signupDate, editProfile }) {
    return (
        <div className="card">
            <div className="card-body">
                <h5 className="card-title">
                    <a href={`/admin/users/${editProfile}`} rel="noopener noreferrer" className="text-dark">{userName}</a>
                </h5>
                <h6 className="card-subtitle mb-2 text-muted">{userEmail}</h6>
                <h6 className="card-subtitle mb-2 text-muted">{`Joined ${signupDate}`}</h6>
            </div>
        </div>
    );
}

export default Profile;

if (document.getElementById('profile')) {
    const container = document.getElementById('profile');
    const root = createRoot(container);
    let userName = document.getElementById('profile').getAttribute('userName');
    let userEmail = document.getElementById('profile').getAttribute('userEmail');
    let signupDate = document.getElementById('profile').getAttribute('signupDate');
    let editProfile = document.getElementById('profile').getAttribute('editProfile');
    root.render(<Profile userName={userName} userEmail={userEmail} signupDate={signupDate} editProfile={editProfile}/>);
}
