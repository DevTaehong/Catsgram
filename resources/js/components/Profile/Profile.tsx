import React from 'react';
import { createRoot } from 'react-dom/client';

type ProfileProps = {
    userName: string
    userEmail: string
    signupDate: string
    editProfile: string | null
}

function Profile({ userName, userEmail, signupDate, editProfile } : ProfileProps) {
    return (
        <div className="card">
            <div className="card-body">
                <h5 className="card-title">
                    <a href={`/admin/users/${editProfile}`} rel="noopener noreferrer" className="text-dark">{userName}</a>
                </h5>
                <h6 className="card-subtitle mb-2 text-muted">{userEmail}</h6>
                <h6 className="card-subtitle mb-2 text-muted">{`Joined ${signupDate}`}</h6>
            </div>
        </div>
    );
}

export default Profile;

if (document.getElementById('profile')) {
    const container: HTMLElement | null = document.getElementById('profile');

    if(container) {
        const userName: string | null = container.getAttribute('userName');
        const userEmail: string | null = container.getAttribute('userEmail');
        const signupDate: string | null = container.getAttribute('signupDate');
        const editProfile: string | null = container.getAttribute('editProfile');
        const root = createRoot(container);
        if(userName && userEmail && editProfile && signupDate) {
            root.render(<Profile userName={userName} userEmail={userEmail} signupDate={signupDate} editProfile={editProfile}/>);
        }
    }
}
