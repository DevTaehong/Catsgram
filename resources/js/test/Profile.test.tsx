/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer';
import Profile from '../components/Profile';
import React from 'react';

it('renders correctly', () => {
    const tree = renderer
    .create(<Profile userName={'Taehong'} userEmail={'minth1123@icloud.com'} signupDate={'Today'} editProfile={null} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
})