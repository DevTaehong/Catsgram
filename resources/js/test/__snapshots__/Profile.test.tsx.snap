// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      <a
        className="text-dark"
        href="/admin/users/null"
        rel="noopener noreferrer"
      >
        Taehong
      </a>
    </h5>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
      minth1123@icloud.com
    </h6>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
      Joined Today
    </h6>
  </div>
</div>
`;
